src/Networking/v2/Models/Network.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace OpenStack\Networking\v2\Models;
+
+use OpenStack\Common\Resource\AbstractResource;
+use OpenStack\Common\Resource\Listable;
+use OpenStack\Common\Resource\Retrievable;
+
+/**
+ * Represents a Network v2 Network.
+ *
+ * @property \OpenStack\Networking\v2\Api $api
+ */
+class Network extends AbstractResource implements Listable, Retrievable
+{
+    public $id;
+    public $name;
+    public $shared;
+    public $status;
+    public $subnets;
+    public $adminStateUp;
+
+    protected $aliases = [
+        'admin_state_up' => 'adminStateUp',
+    ];
+
+    protected $resourceKey = 'network';
+    protected $resourcesKey = 'networks';
+
+    /**
+     * {@inheritDoc}
+     */
+    public function retrieve()
+    {
+        $response = $this->execute($this->api->getNetwork(), ['id' => (string) $this->id]);
+        $this->populateFromResponse($response);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param array $userOptions {@see \OpenStack\Networking\v2\Api::postNetworks}
+     */
+    public function bulkCreate(array $userOptions)
+    {
+        $response = $this->execute($this->api->postNetworks(), [
+            'networks' => $userOptions,
+        ]);
+        $body = $response->json();
+        $json = $body['networks'];
+
+        $networks = [];
+        foreach($json as $resourceData) {
+            $resource = $this->newInstance();
+            $resource->populateFromArray($resourceData);
+            $networks[] = $resource;
+        }
+
+        return $networks;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param array $userOptions {@see \OpenStack\Networking\v2\Api::postNetwork}
+     */
+    public function create(array $userOptions)
+    {
+        $response = $this->execute($this->api->postNetwork(), $userOptions);
+        return $this->populateFromResponse($response);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function update()
+    {
+        $response = $this->execute($this->api->putNetwork(), $this->getAttrs(['id', 'name', 'shared', 'adminStateUp']));
```

Comment:
I added a new method called [`executeWithState`](https://github.com/php-opencloud/openstack/blob/master/src/Common/Resource/AbstractResource.php#L172-L176) that will look at what params the API definition wants, and then populate an array of those keys based off the current object's state. It basically means we can delegate to this method and not have to manually define the update keys. So you'd call like this:

```php
$response = $this->executeWithState($this->api->putNetwork());
return $this->populateFromResponse($response);
```